lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2727,6 +2727,11 @@ void ByteCodeGenerator::EmitFunctionBody(FuncInfo *funcInfo)
     }
     Assert(!pnode->CapturesSyms());
     EmitTopLevelStatement(pnode, funcInfo, false);
+
+    if (funcInfo->IsClassConstructor())
+    {
+        funcInfo->GetParsedFunctionBody()->SetHasNoExplicitReturnValue(false);
```

Comment:
Possibly. Note other calls to SetHasNoExplicitReturnValue() won't happen till inside EmitTopLevelStatement(). I think it's more consistent to set the flag here.

---
In reply to: [78957170](https://github.com/Microsoft/ChakraCore/pull/1577#discussion_r78957170) [](ancestors = 78957170)